### PR TITLE
azurerm_iothub - Add support for cloud_to_device block 

### LINKED ISSUE #13103: Support for cloudToDevice and messagingEndpoints properties for azurerm_iothub
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Add support for cloudToDevice and messagingEndpoints blocks for azurerm_iothub resource. Documentation for those properties can be found [here](https://docs.microsoft.com/en-us/azure/templates/microsoft.devices/iothubs?tabs=json#iothubproperties-object).

### New or Affected Resource(s)

* azurerm_iothub

### Potential Terraform Configuration

```hcl
resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "West Europe"
}

resource "azurerm_iothub" "example" {
  name                = "Example-IoTHub"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location

  sku {
    name     = "S1"
    capacity = "1"
  }

  cloud_to_device { # <---------------------------- This field here
    ...
  }

  messaging_endpoints { # <------------------------ This field here
    ...
  }
}
```

### PR DESCRIPTION
**Pull request by catriona-m**
partially fixes #13103

### CODE REVIEW

File: website/docs/r/iothub.html.markdown [hunk 1]
```diff
@@ -250,6 +262,26 @@ A `file_upload` block supports the following:
...
+
+* `default_ttl` - (Optional) The default time to live for cloud-to-device messages, specified as an [ISO 8601 timespan duration](https://en.wikipedia.org/wiki/ISO_8601#Durations). This value must be between 1 minute and 48 hours, and evaluates to 'PT1H' by default.
+
+* `feedback` - (Optional) A `feedback` block as defined below.
+
+---
+
+A `feedback` block supports the following:
+
+* `time_to_live` - (Optional) The retention time for service-bound feedback messages, specified as an [ISO 8601 timespan duration](https://en.wikipedia.org/wiki/ISO_8601#Durations). This value must be between 1 minute and 48 hours, and evaluates to 'PT1H' by default.
```
Comment (katbyte) [inline website/docs/r/iothub.html.markdown:279]:
```suggestion
* `time_to_live` - (Optional) The retention time for service-bound feedback messages, specified as an [ISO 8601 timespan duration](https://en.wikipedia.org/wiki/ISO_8601#Durations). This value must be between `1` minute and `48` hours, and evaluates to 'PT1H' by default.
```

File: website/docs/r/iothub.html.markdown [hunk 2]
```diff
@@ -250,6 +262,26 @@ A `file_upload` block supports the following:
...
+
+* `max_delivery_count` - (Optional) The maximum delivery count for the feedback queue. This value must be between 1 and 100, and evaluates to 10 by default.
```
Comment (katbyte) [inline website/docs/r/iothub.html.markdown:281]:
```suggestion
* `max_delivery_count` - (Optional) The maximum delivery count for the feedback queue. This value must be between `1` and `100`, and evaluates to `10` by default.
```

**Review by katbyte — APPROVED** [review]
Thanks @catriona-m - aside from a couple comments I've left inline LGTM 💯

File: website/docs/r/iothub.html.markdown [hunk 3]
```diff
@@ -250,6 +262,26 @@ A `file_upload` block supports the following:
 
 * `max_delivery_count` - (Optional) The number of times the IoT hub attempts to deliver a file upload notification message. It evaluates to 10 by default.
 
+---
+
+A `cloud_to_device` block supports the following:
+
+* `max_delivery_count` - (Optional) The maximum delivery count for cloud-to-device per-device queues. This value must be between 1 and 100, and evaluates to 10 by default.
+
+* `default_ttl` - (Optional) The default time to live for cloud-to-device messages, specified as an [ISO 8601 timespan duration](https://en.wikipedia.org/wiki/ISO_8601#Durations). This value must be between 1 minute and 48 hours, and evaluates to 'PT1H' by default.
```
[the remark this answers is not in the document]
Comment (catriona-m) [reply website/docs/r/iothub.html.markdown]: thanks @katbyte I wasn't sure if I should highlight 1 and 48 here as they aren't actual possible values in this case?

**Pull request merged**